packages/contracts/src/RelayHub.sol

```diff
@@ -176,16 +180,21 @@ contract RelayHub is IRelayHub, Ownable {
     {
         RelayCallData memory vars;
         vars.initialGasLeft = aggregateGasleft();
+        vars.relayRequestId = GsnUtils.getRelayRequestID(relayRequest, signature);
         require(!isDeprecated(), "hub deprecated");
         vars.functionSelector = relayRequest.request.data.length>=4 ? MinLibBytes.readBytes4(relayRequest.request.data, 0) : bytes4(0);
-        require(msg.sender == tx.origin, "relay worker must be EOA");
-        vars.relayManager = workerToManager[msg.sender];
-        require(vars.relayManager != address(0), "Unknown relay worker");
-        require(relayRequest.relayData.relayWorker == msg.sender, "Not a right worker");
-        require(
-            isRelayManagerStaked(vars.relayManager),
-            "relay manager not staked"
-        );
+        if (msg.sender == batchGateway){
+            require(signature.length == 0, "batch gateway signature not zero");
```

Comment:
remove this zero-length sig check: you prevent batching of ECDSA requests. the required check is below (non-empty signature UNLESS its a batch gateway)

Reply:
^^

Reply:
Fixed.